File: 1957. Delete Characters to Make Fancy String/solution.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        result = []
        for c in s:
            n = len(result)
            if n < 2 or not (result[-1] == c and result[-2] == c):
                result.append(c)
        return ''.join(result)

File: 1957. Delete Characters to Make Fancy String/solution.cpp
#include <string>
using namespace std;

class Solution {
public:
    string makeFancyString(string s) {
        string result;
        for (char c : s) {
            int n = result.size();
           
            if (n < 2 || !(result[n - 1] == c && result[n - 2] == c)) {
                result.push_back(c);
            }
        }
        return result;
    }
};